manim/mobject/types/vectorized_mobject.py

```diff
@@ -482,24 +489,24 @@ def add_smooth_curve_to(self, *points):
             ])
         return self
 
-    def has_new_path_started(self):
+    def has_new_path_started(self) -> bool:
         nppcc = self.n_points_per_cubic_curve  # 4
         return len(self.points) % nppcc == 1
 
-    def get_last_point(self):
+    def get_last_point(self) -> Vector:
         return self.points[-1]
 
-    def is_closed(self):
+    def is_closed(self) -> bool:
         return self.consider_points_equals(
             self.points[0], self.points[-1]
         )
 
-    def add_points_as_corners(self, points):
+    def add_points_as_corners(self, points: List[Vector]) -> List[Vector]:
         for point in points:
             self.add_line_to(point)
         return points
```

Comment:
we do not require it to be a list, only an iterable - i.e., we only need to be able to iterate over it, so we should be allowed to specify a list, but also a tuple, a zip, etc. However, it will return the own type specified, so we have two alternatives here. Either:
```suggestion
    def add_points_as_corners(self, points: typing.Iterable[Vector]) -> List[Vector]:
        for point in points:
            self.add_line_to(point)
        return list(points)
```
or (for non-breaking behavior):
```suggestion
    IV = typing.TypeVar("IV", bound=typing.Iterable[Vector])  # perhaps add this to the top of the file
    def add_points_as_corners(self, points: IV) -> IV:
        for point in points:
            self.add_line_to(point)
        return points
```